lib/reporter/stats-reporter.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const colors = require('colors');
+const filesize = require('filesize');
+const Logger = require('../logger');
+const percent = require('calc-percent');
+const program = require('commander');
+const Table = require('cli-table');
+const uniq = require('lodash/uniq');
+
+const logger = new Logger();
+
+function statsReporter(statsObj) {
+	logger.updateLevel(program.logLevel);
+	const sizesPerType = getSizesPerType(statsObj.src, statsObj.dest);
+	const tableData = calculateSavingsPerType(sizesPerType);
+	const table = generateTable(tableData);
+
+	logger.log(table.toString());
+}
+
+function getSizesPerType(srcSizes, destSizes) {
+	const srcTypes = Object.keys(srcSizes);
+	const destTypes = Object.keys(destSizes);
+
+	return uniq(srcTypes.concat(destTypes)).reduce((sizes, type) => {
+		if (!srcSizes[type] || !destSizes[type]) return sizes;
+
+		sizes[type] = {
+			src: srcSizes[type],
+			dest: destSizes[type]
+		}
+
+		return sizes;
+	}, {});
+}
+
+function calculateSavingsPerType(sizes) {
+	return Object.keys(sizes).reduce((collection, type) => {
+		const savedBytes = sizes[type].src - sizes[type].dest;
+		collection[type] = Object.assign(sizes[type], { saving: savedBytes });
+
+		return collection;
+	}, {});
+}
+
+function generateTable(tableData) {
```

Comment:
consistent naming: `createTable(data)`, `createTableRow(...)`

I also think there should be a function `createTableRow` (singular) instead of `getTableRows` (plural), so the `createTable` function would iterate through the data and call `createTableRow` for each.